buzsaki_lab_to_nwb/peyrache_code/peyrachemiscdatainterface.py

```diff
@@ -0,0 +1,126 @@
+"""Authors: Cody Baker and Ben Dichter."""
+import numpy as np
+from pathlib import Path
+from scipy.io import loadmat
+from warnings import warn
+import pandas as pd
+
+from nwb_conversion_tools.basedatainterface import BaseDataInterface
+from pynwb import NWBFile
+from pynwb.file import TimeIntervals
+from pynwb.behavior import SpatialSeries, Position
+from hdmf.backends.hdf5.h5_utils import H5DataIO
+from spikeextractors import NeuroscopeRecordingExtractor
+
+from ..neuroscope import get_events, check_module
+
+
+def peyrache_spatial_series(name: str, description: str, data: np.array, conversion: float, pos_sf: float = 1250 / 32):
+    """Specific constructor for Peyrache style spatial series."""
+    return SpatialSeries(
+        name=name,
+        description=description,
+        data=H5DataIO(data, compression="gzip"),
+        conversion=conversion,
+        reference_frame="Unknown",
+        starting_time=0.,
+        rate=pos_sf,
+        resolution=np.nan
+    )
+
+
+class PeyracheMiscInterface(BaseDataInterface):
+    """Primary data interface for miscellaneous aspects of the PeyracheA dataset."""
+
+    @classmethod
+    def get_source_schema(cls):
+        return dict(properties=dict(folder_path=dict(type="string")))
```

Comment:
since this should return a proper json schema, there will be some relevant fields missing in this implementation.
check out this example: https://github.com/catalystneuro/jaeger-lab-to-nwb/blob/2bf42b97a32873d27d301c76edb42fce0c63f07d/jaeger_lab_to_nwb/bpodconverter/bpoddatainterface.py#L11-L28

Reply:
The example you linked begins with the comment "Return a partial JSON schema indicating the input arguments and their types.", which is exactly what this is (but even more partial).

Excluding the required field does not seem to effect anything downstream, and what is listed here passes Draft7Validation without issue.  If we wanted to enforce full json schemas we should probably do it on the nwb-conversion-tools side so it isn't optional for custom interfaces like this one, or for the jaeger one.

But also, why would it have to be a full schema? Do you need it for some kind of form thing?